### PR TITLE
Remove indefinite wait for ACM Pods

### PR DESCRIPTION
**Pull request by dhaiducek**
Use `kubectl wait` against pods labeled `app in (grc,policyreport,multicluster-operators-standalone-subscription)` rather than waiting for all pods (so that a flaky non-GRC pods don't block our tests from running).

Also bumps the propagator version and cleans up some Shellcheck identified issues.

### CODE REVIEW

**Comment by dhaiducek** [comment]
/override ci/prow/test-e2e-grc-framework 

Failed as expected at `Expected metrics for config_policy_evaluation_seconds_total`

**Comment by dhaiducek** [comment]
/retest

I'm not sure why this is failing:
```
 GRC: [P1][Sev1][policy-grc] Test install Operator When all defaults are used on the subscription test-op-47229 should be created and become compliant [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:459
  STEP: Verifying that the policy policy-test/test-op-47229 is Compliant @ 08/02/24 23:16:39.396
  [FAILED] in [It] - /go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:463 @ 08/02/24 23:18:39.397
  STEP: Cleaning up the operator installation @ 08/02/24 23:18:39.397
  STEP: Verifying that the policy policy-test/cleanup-grcqeoptest-ns is Compliant @ 08/02/24 23:18:41.406
• [FAILED] [123.480 seconds]
GRC: [P1][Sev1][policy-grc] Test install Operator When all defaults are used on the subscription [It] test-op-47229 should be created and become compliant [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:459
  [FAILED] Timed out after 120.001s.
  The function passed to Eventually failed at /go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:897 with:
  Expected
      <string>: NonCompliant
  to equal
      <string>: Compliant
  In [It] at: /go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:463 @ 08/02/24 23:18:39.397 
```

**Comment by dhaiducek** [comment]
Seems to be either that or this one:
```
 GRC: [P1][Sev1][policy-grc] Test the compliance history API Creates a policy with a noncompliant and compliant operator policy [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:714
  STEP: Creating the namespace @ 08/02/24 22:08:04.914
  STEP: Verifying that the policy has created the namespace @ 08/02/24 22:08:05.639
  STEP: Verifying that the policy open-cluster-management-global-set/ch-operator-prereq-policy is Compliant @ 08/02/24 22:08:05.639
  STEP: Creating the Policy @ 08/02/24 22:08:06.649
  STEP: Verifying that the policy open-cluster-management-global-set/op-compliance-api is NonCompliant @ 08/02/24 22:08:07.251
  STEP: Verifying that there are NonCompliant compliance events for the Operator parent policy @ 08/02/24 22:08:08.259
  STEP: Patch operator Policy to be Compliant @ 08/02/24 22:08:08.274
  STEP: The Policy should be Compliant on all clusters @ 08/02/24 22:08:09.66
  [FAILED] in [It] - /go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:834 @ 08/02/24 22:09:09.661
  STEP: Deleting the policy-uninstall-gk policy @ 08/02/24 22:09:09.661
  STEP: Deleting Postgres @ 08/02/24 22:09:09.742
  STEP: Deleting the service account in the default namespace @ 08/02/24 22:09:09.894
  STEP: Unsetting the governance-policy-framework ClusterManagementAddOn to use the AddOnDeploymentConfig @ 08/02/24 22:09:09.997
  STEP: Deleting the AddOnDeploymentConfig @ 08/02/24 22:09:10.008
  Cluster local-cluster current compliance message: NonCompliant; the policy spec is valid, the OperatorGroup matches what is required by the policy, the Subscription matches what is required by the policy, no InstallPlans requiring approval were found, ClusterServiceVersion (quay-operator.v3.8.15) - intersecting operatorgroups provide the same apis, there are CRDs present for the operator, no existing operator Deployments, CatalogSource was found
  Cluster managed-1 current compliance message: Compliant; the policy spec is valid, the OperatorGroup matches what is required by the policy, the Subscription matches what is required by the policy, no InstallPlans requiring approval were found, ClusterServiceVersion (quay-operator.v3.8.15) - install strategy completed with no errors, there are CRDs present for the operator, all operator Deployments have their minimum availability, CatalogSource was found
  STEP: Deleting the operator policy @ 08/02/24 22:09:10.013
  STEP: Deleting the namespace policy @ 08/02/24 22:09:10.104
• [FAILED] [65.293 seconds]
GRC: [P1][Sev1][policy-grc] Test the compliance history API [It] Creates a policy with a noncompliant and compliant operator policy [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:714
  [FAILED] Timed out after 60.001s.
  The function passed to Eventually failed at /go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:897 with:
  Expected
      <string>: NonCompliant
  to equal
      <string>: Compliant
  In [It] at: /go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:834 @ 08/02/24 22:09:09.661 
```

**Comment by dhaiducek** [comment]
/retest

```
 GRC: [P1][Sev1][policy-grc] Test install Operator When all defaults are used on the subscription test-op-47229 should be created and become compliant [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:459
  STEP: Verifying that the policy policy-test/test-op-47229 is Compliant @ 08/03/24 02:04:20.113
  [FAILED] in [It] - /go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:463 @ 08/03/24 02:06:20.114
  STEP: Cleaning up the operator installation @ 08/03/24 02:06:20.114
  STEP: Verifying that the policy policy-test/cleanup-grcqeoptest-ns is Compliant @ 08/03/24 02:06:20.686
• [FAILED] [122.202 seconds]
GRC: [P1][Sev1][policy-grc] Test install Operator When all defaults are used on the subscription [It] test-op-47229 should be created and become compliant [BVT]
/go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:459
  [FAILED] Timed out after 120.001s.
  The function passed to Eventually failed at /go/src/github.com/stolostron/governance-policy-framework/test/integration/compliance_history_test.go:897 with:
  Expected
      <string>: NonCompliant
  to equal
      <string>: Compliant
  In [It] at: /go/src/github.com/stolostron/governance-policy-framework/test/integration/policy_install_operator_test.go:463 @ 08/03/24 02:06:20.114 
```

File: build/patch-cluster-prow.sh [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
-  sleep 5
-done
+acm_installed_namespace=$(oc get subscriptions.operators.coreos.com --all-namespaces | grep advanced-cluster-management | awk '{print $1}')
 
-./build/patch-dev-images.sh
+echo "* Initial Pod state:"
+oc -n "${acm_installed_namespace}" get pods
+
+echo "* Waiting for prerequisite pods:"
+kubectl wait pod -l 'app in (grc,policyreport,multicluster-operators-standalone-subscription)' \
```
Comment (yiraeChristineKim) [inline build/patch-cluster-prow.sh:12]: Is this work?  I thought this one work `$(app in (grc,policyreport,multicluster-operators-standalone-subscription) `

Comment (dhaiducek) [reply build/patch-cluster-prow.sh:12]: The `$(` notation is for a subshell, but this isn't a subshell command (unless I'm misunderstanding your question).

Comment (dhaiducek) [reply build/patch-cluster-prow.sh:12]: It's a string meant to be translated by the `-l` flag for a MatchExpressions.

Comment (yiraeChristineKim) [reply build/patch-cluster-prow.sh:12]: oh I didn't know good to know!

**Review by yiraeChristineKim — APPROVED** [review]

File: build/patch-cluster-prow.sh (hunk 1 above)
Comment (dhaiducek) [reply build/patch-cluster-prow.sh:12]: For reference:
- https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#equality-based-requirement
- https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement

**Pull request merged**